test/unittests/api_test.cpp

```diff
@@ -11,6 +11,146 @@
+      (func (import "mod1" "foo2") (param i32) (result i32))
+      (func (import "mod2" "foo1") (param i32) (result i32))
+      (func (import "mod2" "foo2") (param i64))
+      (global (import "mod1" "g") i32) ;; just to test combination with other import types
+    */
+    const auto wasm = from_hex(
+        "0061736d01000000010e036000017f60017f017f60017e00023b05046d6f643104666f6f310000046d6f643104"
+        "666f6f320001046d6f643204666f6f310001046d6f643204666f6f320002046d6f64310167037f00");
+    const auto module = parse(wasm);
+
+    std::vector<ImportedFunction> imported_functions = {
+        {"mod1", "foo1", {}, ValType::i32, function_returning_value(0)},
+        {"mod1", "foo2", {ValType::i32}, ValType::i32, function_returning_value(1)},
+        {"mod2", "foo1", {ValType::i32}, ValType::i32, function_returning_value(2)},
+        {"mod2", "foo2", {ValType::i64}, std::nullopt, function_returning_void},
+    };
+
+    const auto external_functions =
+        resolve_imported_functions(module, std::move(imported_functions));
+
+    EXPECT_EQ(external_functions.size(), 4);
+
+    uint64_t global = 0;
+    const std::vector<ExternalGlobal> external_globals{{&global, false}};
+    auto instance = instantiate(
+        module, external_functions, {}, {}, std::vector<ExternalGlobal>(external_globals));
+
+    EXPECT_THAT(execute(*instance, 0, {}), Result(0));
+    EXPECT_THAT(execute(*instance, 1, {}), Result(1));
+    EXPECT_THAT(execute(*instance, 2, {}), Result(2));
+    EXPECT_THAT(execute(*instance, 3, {}), Result());
+
+
+    std::vector<ImportedFunction> imported_functions_reordered = {
+        {"mod2", "foo1", {ValType::i32}, ValType::i32, function_returning_value(2)},
+        {"mod1", "foo2", {ValType::i32}, ValType::i32, function_returning_value(1)},
+        {"mod1", "foo1", {}, ValType::i32, function_returning_value(0)},
+        {"mod2", "foo2", {ValType::i64}, std::nullopt, function_returning_void},
+    };
+
+    const auto external_functions_reordered =
+        resolve_imported_functions(module, std::move(imported_functions_reordered));
+    EXPECT_EQ(external_functions_reordered.size(), 4);
+
+    auto instance_reordered = instantiate(module, external_functions_reordered, {}, {},
+        std::vector<ExternalGlobal>(external_globals));
+
+    EXPECT_THAT(execute(*instance_reordered, 0, {}), Result(0));
+    EXPECT_THAT(execute(*instance_reordered, 1, {}), Result(1));
+    EXPECT_THAT(execute(*instance_reordered, 2, {}), Result(2));
+    EXPECT_THAT(execute(*instance_reordered, 3, {}), Result());
+
+
+    std::vector<ImportedFunction> imported_functions_extra = {
+        {"mod1", "foo1", {}, ValType::i32, function_returning_value(0)},
+        {"mod1", "foo2", {ValType::i32}, ValType::i32, function_returning_value(1)},
+        {"mod2", "foo1", {ValType::i32}, ValType::i32, function_returning_value(2)},
+        {"mod2", "foo2", {ValType::i64}, std::nullopt, function_returning_void},
```

Comment:
Can you add one test case somewhere which takes 2 inputs?

Reply:
Made this void function take 2 parameters